commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -18,23 +19,75 @@
 
 
 import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
 
 public class MwVolleyApi {
 
     private static RequestQueue REQUEST_QUEUE;
     private static final Gson GSON = new GsonBuilder().create();
     private Context context;
+    private static String coordsLog;
+
+    protected static HashSet<String> categorySet;
```

Comment:
Done, ditto with the List/ArrayList issue. I guess as a general rule of thumb it is better to use the more general type?